google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -2154,6 +2157,80 @@ public ApiFuture<Empty> asyncClose() {
     }
   }
 
+  private final class MultiplexedSessionWaiterFuture
+      extends ForwardingListenableFuture<MultiplexedSession> {
+
+    private final Duration MAX_MULTIPLEXED_SESSION_WAIT_TIMEOUT = Duration.ofSeconds(60);
+    private final SettableFuture<MultiplexedSession> waiter = SettableFuture.create();
+
+    @Override
+    protected ListenableFuture<? extends MultiplexedSession> delegate() {
+      return waiter;
+    }
+
+    private void put(MultiplexedSession session) {
+      waiter.set(session);
+    }
+
+    private void put(SpannerException e) {
+      waiter.setException(e);
+    }
+
+    @Override
+    public MultiplexedSession get() {
+      ISpan span = tracer.spanBuilder(WAIT_FOR_MULTIPLEXED_SESSION);
+      try (IScope waitScope = tracer.withSpan(span)) {
+        final Duration timeout =
+            MoreObjects.firstNonNull(
+                options.getAcquireSessionTimeout(), MAX_MULTIPLEXED_SESSION_WAIT_TIMEOUT);
```

Comment:
Note that:
1. We will only enter a waiting state if the initial `CreateSession` call is still in flight. In all other cases, a multiplexed session will be available.
2. MAX_MULTIPLEXED_SESSION_WAIT_TIMEOUT is 60s.
3. The default timeout for the `CreateSession` RPC is 30s.
4. So the `CreateSession` RPC will timeout before this will timeout, unless the user has modified the default timeout value for `CreateSession`.

Reply:
`Done, simplified this flow by using AtomicReference<SettableApiFuture<MultiplexedSessionFuture>>.`